store/tikv/region_cache.go

```diff
@@ -301,16 +309,14 @@ func (c *RegionCache) GetTiKVRPCContext(bo *Backoffer, id RegionVerID, replicaRe
 	}
 
 	regionStore := cachedRegion.getStore()
-	var store *Store
-	var peer *metapb.Peer
-	var storeIdx int
-	switch replicaRead {
-	case kv.ReplicaReadFollower:
-		store, peer, storeIdx = cachedRegion.FollowerStorePeer(regionStore, followerStoreSeed)
-	default:
-		store, peer, storeIdx = cachedRegion.WorkStorePeer(regionStore)
-	}
-	addr, err := c.getStoreAddr(bo, cachedRegion, store, storeIdx)
+
+	candidate := optimizeAvailableStores(regionStore, engine)
```

Comment:
I prefer to choose the best one in `cachedRegion.ChooseStorePeer`, because we use `seed` for follower read. Sort peers here maybe can't help.

Reply:
it's not only sort, we can do filter in here :smile: 

for example, we want to "strict" isolated read: appA using 1, 2, 3 and appB using 4, 5, 6, appA never access appB's node even if all 1, 2, 3 having down.

emmm, we still need address idx manager problem, but here just want to limit a small list.

at last, if we can remove seed, sort at this will simplify retry logic..and leader read maybe can get benefit from list, for example: 1, 2, 3 in bj, 4, 5 in xian, when leader in 3 send fail or NotLeader with no leader, we should better switch to 1, 2, before try 4, 5 in xian?